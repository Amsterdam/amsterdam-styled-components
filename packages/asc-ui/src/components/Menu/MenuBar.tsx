import React from 'react'
import ButtonBar, { Props as MenuBarProps } from '../ButtonBar'
import MenuStyle from '../../styles/components/MenuStyle'

const { MenuBarStyle } = MenuStyle

export type Props = MenuBarProps

const MenuBar: React.FC<Props> = ({ children, styledComponent, ...otherProps }) => (
  <ButtonBar styledComponent={styledComponent} {...otherProps}>{children}</ButtonBar>
)

MenuBar.defaultProps = {
  styledComponent: MenuBarStyle
}

export default MenuBar
