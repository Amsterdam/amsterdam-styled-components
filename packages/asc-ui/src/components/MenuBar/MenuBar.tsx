import React from 'react'
import styled from '../../styled-components'
import { color, breakpoint, focusStyle, fromTheme } from '../../styles/utils'

const MenuButtonStyle = styled.button<{}>`
  ${focusStyle()}
  display: flex;
  background-color: transparent;
  border: 0px;
  align-items: center;
  padding: 0;
`

const MenuButton = ({ id, label, ...otherProps }: any) => {
  return (
    <MenuButtonStyle {...{ id }} {...otherProps}>
      <span>{label && label}</span>
    </MenuButtonStyle>
  )
}

const MenuBarItemStyle = styled.li`
  ${MenuButtonStyle} {
    font-family: 'AvenirNextLTW01-Medium';
    padding: 15px;
    cursor: pointer;

    & > span {
      color: ${color('tint', 'level6')};
      line-height: 22px;
      border: 0;
      margin: 0;
      border-bottom: 2px transparent solid;
      background-color: transparent;
    }

    &:hover > span,
    &:focus > span {
      color: ${color('tint', 'level6')};
      line-height: 22px;
      border-bottom: 2px solid red;
      outline: none;
    }
  }

  &:hover,
  &:focus {
    outline: none;
    background-color: ${color('tint', 'level2')};
  }
`

export const MenuBarItem: React.FC<{
  label?: string
  onClick?: any
  component?: React.Component
}> = ({ label, onClick, component, ...otherProps }) => (
  <MenuBarItemStyle {...otherProps}>
    {component || <MenuButton id="1" label={label} onClick={onClick} />}
  </MenuBarItemStyle>
)

type MenuBarStyleProps = {}
type Props = MenuBarStyleProps

const MenuBarStyle = styled.ul`
  display: flex;
  flex-direction: row;
  height: 100%;
  margin: 0;
  padding: 0;
  list-style: none;
  font-size: ${fromTheme('typography.fontSize')};
  /* cursor: pointer; */

  &:focus {
    outline: none;
  }

  @media screen and ${breakpoint('max-width', 'laptop')} {
    display: none;
  }
`

const MenuBar: React.FC<Props> = ({ children, ...otherProps }) => (
  <MenuBarStyle {...otherProps}>{children}</MenuBarStyle>
)

export default MenuBar
