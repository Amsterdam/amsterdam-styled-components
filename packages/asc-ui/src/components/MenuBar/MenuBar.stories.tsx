import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import MenuBar, { MenuBarItem } from './MenuBar'
import { ThemeProvider } from '../../index'

const MenuBarStory: React.FC<{}> = () => (
  <ThemeProvider>
      <MenuBar>
        <MenuBarItem label="MenuItem1" onClick={action('click on MenuItem1')} />
        <MenuBarItem label="MenuItem2" onClick={action('click on MenuItem2')} />
      </MenuBar>
  </ThemeProvider>
)

storiesOf('Atoms/MenuBar', module).add('default state', () => <MenuBarStory />)
