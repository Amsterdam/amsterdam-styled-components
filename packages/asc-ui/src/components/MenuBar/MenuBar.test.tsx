import * as React from 'react'
import { shallow } from 'enzyme'
import MenuBar, { MenuBarItem } from './MenuBar'

describe('MenuBar', () => {
  const mockOnClick = jest.fn()
  const children = (
    <>
      <MenuBarItem label="Inhoud" onClick={mockOnClick} />
      <MenuBarItem label="Over" onClick={mockOnClick} />
    </>
  )
  const component = shallow(<MenuBar>{children}</MenuBar>)
  const menuBarItem = component.find(MenuBarItem)

  it('should render with the correct children', () => {
    expect(component.props().children).toBe(children)
  })

  it('should render with the correct label', () => {
    expect(menuBarItem.at(0).props().label).toBe('Inhoud')
    expect(menuBarItem.at(1).props().label).toBe('Over')
  })

  it('should handle the onclick event', () => {
    menuBarItem.at(0).simulate('click')

    expect(mockOnClick).toHaveBeenCalled()
  })
})
