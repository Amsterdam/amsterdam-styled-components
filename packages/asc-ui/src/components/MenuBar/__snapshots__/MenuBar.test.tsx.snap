// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar should render 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  border: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
}

.c3:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: medium;
}

.c1 .c2 {
  font-family: 'AvenirNextLTW01-Medium';
  padding: 15px;
  cursor: pointer;
}

.c1 .c2 > span {
  color: #323232;
  line-height: 22px;
  border: 0;
  margin: 0;
  border-bottom: 2px transparent solid;
  background-color: transparent;
}

.c1 .c2:hover > span,
.c1 .c2:focus > span {
  color: #323232;
  line-height: 22px;
  border-bottom: 2px solid red;
  outline: none;
}

.c1:hover,
.c1:focus {
  outline: none;
  background-color: #F5F5F5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 100%;
  margin: 0;
  padding: 0;
  list-style: none;
  font-size: 16px;
}

.c0:focus {
  outline: none;
}

@media screen and (max-width:1024px) {
  .c0 {
    display: none;
  }
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <button
      class="c2 c3"
      id="1"
    >
      <span>
        Inhoud
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      class="c2 c3"
      id="1"
    >
      <span>
        Over
      </span>
    </button>
  </li>
</ul>
`;
